src/components/LoginForm..jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react';
+import {Tab, Tabs, TextField, Stack, Button, Container } from "@mui/material";
+import { Box } from '@mui/system';
+
+const passwordMatchingText = {
+    error: "Passwords do not match",
+    valid: ""
+}
+function LoginForm() {
+    const [currentTabIndex, setCurrentTabIndex] = useState(0);
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [passwordVerification, setPasswordVerification] = useState('');
+    const [passwordMatchingMsg, setPasswordMatchingMsg] = useState(passwordMatchingText.valid)
+    
+    const handleTabChange = (event, tabIndex) => {
+        setCurrentTabIndex(tabIndex);
+        setEmail('')
+        setPassword('')
+        setPasswordVerification('')
+        setPasswordMatchingMsg(passwordMatchingText.valid)
+    };
+
+    const handleEmailChange = (event) => {
+        setEmail(event.target.value)
+    }
+
+    const handlePasswordChange = (event) => {
+        setPassword(event.target.value)
+        setPasswordMatchingMsg(passwordMatchingText.valid)
+    }
+
+    const handlePasswordVerificationChange = (event) => {
+        setPasswordVerification(event.target.value)
+        setPasswordMatchingMsg(passwordMatchingText.valid)
+    }
+
+    /*
+    TODO:
+    - Handle api status code (200 vs != 200)
+    */
+    const onLogin = (event) => {
+        event.preventDefault()
+        alert("Welcome " + email)
+
+    }
+
+    /*
+    TODO:
+    - Handle api status code (200 vs != 200)
+    */
+    const onCreateAccount = (event) => {
+        event.preventDefault()
+        if (password !== passwordVerification) {
```

Comment:
This validation works but I'm wondering about the overall strategy: are you thinking of eventually using a library for form validation/displaying error messages, or will we implement it all ourselves?

On the signup form, I think we'll eventually need:

- Email address format validation
- Password validation
     - Number of characters
     - Is it a commonly-used password?
     - Strength/entropy (could use our password strength algorithm here)
     - etc...
     
The backend will be unable to do any initial password validation because it will receive a hash of the password, not the plaintext. Even a blank password (empty string) will hash to a fixed-length Base 64 string that will be indistinguishable from a legit password. I think the frontend will have to do it all.

Reply:
We have another task for validation. I split it out since I thought it'd be easier to create the form first then build the validation afterwards.

Reply:
Yes the front end could handle all the validation so we can handle it there. I did not add any real form validation as there was a discussion to save it next week during our weekly meeting with this ticket:

- https://github.com/secure-password-manager/PasswordManagerWeb/issues/11